planning/behavior_path_planner/autoware_behavior_path_start_planner_module/test/test_geometric_pull_out.cpp

```diff
@@ -0,0 +1,225 @@
+      pull_out_arc_path_interval_;
+    parameters->parallel_parking_parameters.center_line_path_interval = center_line_path_interval_;
+    parameters->th_moving_object_velocity = th_moving_object_velocity_;
+
+    auto time_keeper = std::make_shared<autoware::universe_utils::TimeKeeper>();
+    geometric_pull_out_ =
+      std::make_shared<GeometricPullOut>(*node_, *parameters, lane_departure_checker_, time_keeper);
+  }
+
+  void initialize_planner_data()
+  {
+    planner_data_.parameters.backward_path_length = backward_path_length_;
+    planner_data_.parameters.forward_path_length = forward_path_length_;
+    planner_data_.parameters.wheel_base = vehicle_info_.wheel_base_m;
+    planner_data_.parameters.wheel_tread = vehicle_info_.wheel_tread_m;
+    planner_data_.parameters.front_overhang = vehicle_info_.front_overhang_m;
+    planner_data_.parameters.left_over_hang = vehicle_info_.left_overhang_m;
+    planner_data_.parameters.right_over_hang = vehicle_info_.right_overhang_m;
+  }
+
+  // Parameter variables
+  double lane_departure_check_expansion_margin_{0.0};
+  double pull_out_max_steer_angle_{0.0};
+  double pull_out_arc_path_interval_{0.0};
+  double center_line_path_interval_{0.0};
+  double th_moving_object_velocity_{0.0};
+  double backward_path_length_{0.0};
+  double forward_path_length_{0.0};
+};
+
+TEST_F(TestGeometricPullOut, GenerateValidGeometricPullOutPath)
+{
+  const auto start_pose =
+    geometry_msgs::build<geometry_msgs::msg::Pose>()
+      .position(geometry_msgs::build<geometry_msgs::msg::Point>().x(362.181).y(362.164).z(100.000))
+      .orientation(
+        geometry_msgs::build<geometry_msgs::msg::Quaternion>().x(0.0).y(0.0).z(0.709650).w(
+          0.704554));
+
+  const auto goal_pose =
+    geometry_msgs::build<geometry_msgs::msg::Pose>()
+      .position(geometry_msgs::build<geometry_msgs::msg::Point>().x(365.658).y(507.253).z(100.000))
+      .orientation(
+        geometry_msgs::build<geometry_msgs::msg::Quaternion>().x(0.0).y(0.0).z(0.705897).w(
+          0.708314));
+
+  // Set up current odometry at start pose
+  auto odometry = std::make_shared<nav_msgs::msg::Odometry>();
+  odometry->pose.pose = start_pose;
+  odometry->header.frame_id = "map";
+  planner_data_.self_odometry = odometry;
+
+  // Setup route
+  const auto route = makeBehaviorRouteFromLaneId(
+    4619, 4635, "autoware_test_utils", "road_shoulder/lanelet2_map.osm");
+  route_handler_->setRoute(route);
+
+  // Update planner data with the route handler
+  planner_data_.route_handler = route_handler_;
+  geometric_pull_out_->setPlannerData(std::make_shared<PlannerData>(planner_data_));
+
+  // Plan the pull out path
+  PlannerDebugData debug_data;
+  auto result = plan(start_pose, goal_pose, debug_data);
+
+  // Assert that a valid geometric geometric pull out path is generated
+  ASSERT_TRUE(result.has_value()) << "Geometric pull-out path generation failed.";
+  EXPECT_FALSE(result->partial_paths.empty())
```

Comment:
better to check the size is 2?

Reply:
Thanks! applied in https://github.com/autowarefoundation/autoware.universe/pull/9640/commits/80183423999e4fa15bac3ae10e89362095bfc38e